API.md

```diff
@@ -0,0 +1,63 @@
+# SegM8 API
+
+## `class SegM8`
+
+Create an object of type `SegM8` to communicate with a chain of particular [SegM8 7-segment indicator modules](https://my.amperka.com/modules/SegM8).
+
+### `SegM8(uint8_t pinCS, uint8_t deviceCount = 1)`
+
+Constructs a new SegM8 object that uses the default hardware SPI bus.
+
+- `pinCS`: the chip select (also known as slave select) pin is used to control the shift-register latch. It is marked `CS` on the SegM8 board. Any GPIO pin of the controller may be used.
+- `deviceCount`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines a single module.
+
+### `SegM8(uint8_t pinCS, uint8_t pinDI, uint8_t pinCLK, uint8_t deviceCount = 1)`
+
+Constructs a new SegM8 object that uses a software SPI bus.
+
+- `pinCS`: the chip select (also known as slave select) pin is used to control the shift-register latch. It is marked `CS` on the SegM8 module. Any GPIO pin of the controller may be used.
+- `pinDI`: the data input pin used to send data to the peripherals. It is marked `DI` on the SegM8 module. In SPI terms known as MOSI (Master Out Slave In). Any GPIO pin of the controller may be used.
+- `pinCLK`: the clock pin is used to clock pulses that synchronize data transmission generated by the master. It is marked `CLK` on the SegM8 module. Also known as SCK in SPI terms. Any GPIO pin of the controller may be used.
+- `deviceCount`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines a single module.
+
+The difference between hardware and software SPI is interaction speed. Software SPI runs slower than hardware SPI but hardware SPI works only on specific board pins: `MOSI`, `MISO` (not used in SegM8) and `SCK`.
+
+### `void begin()`
+
+Initializes the given interface, prepares the board for the communication, and sets all module segments to the 'dark' state.
+Call this method before interacting with SegM8. For example, in the `setup()`.
+
+### `void startBatch(void)`
+
+The begin of output transaction. Moves current output pointer to the beginning of the daisy-chain;
+
+### `void display(int number, uint8_t width, uint8_t flags = SEGM8_ALIGN_RIGHT)`
```

Comment:
What about the type of `long`? If the user decides to use it, he will get an error that the method call is ambiguous, because there are several types of type conversion. Or do we assume that the user should not use the `long` type?

Reply:
The power supply of the module is such that more than 6 modules will be difficult to power from a single source. It is necessary to power up groups of segments from different sources, and this is difficult for a dumb user.

Reply:
The user can take a variable of type `long` from some third-party library/example and pass it to the method, which will cause a compilation error. And maybe there is a user who decides to display very large numbers. At least we don't say the opposite anywhere.

Reply:
"maybe there is a user who decides to display very large numbers" .. for example? 
I can’t come up with this. 
Large displays are designed to understand and realize something at a glance, and not to peer at each digit in huge numbers. There are logs for this.

Reply:
The number of seconds since January 1, 1970, or how much time is left until a certain event, or the amount of money in your current account, or the public debt of America :) I agree that these are very rare use cases. But the key point here is that we are not talking about the fact that it’s impossible. I think we need a third opinion. @Matthew3000 @igor605ds What do you think?

Reply:
Done in new version